packages/admin-next/dashboard/src/hooks/api/fulfillment-sets.tsx

```diff
@@ -0,0 +1,172 @@
+import { FetchError } from "@medusajs/js-sdk"
+import { HttpTypes } from "@medusajs/types"
+import {
+  QueryKey,
+  UseMutationOptions,
+  UseQueryOptions,
+  useMutation,
+  useQuery,
+} from "@tanstack/react-query"
+import { sdk } from "../../lib/client"
+import { queryClient } from "../../lib/query-client"
+import { queryKeysFactory } from "../../lib/query-key-factory"
+import { shippingOptionsQueryKeys } from "./shipping-options"
+import { stockLocationsQueryKeys } from "./stock-locations"
+
+const FULFILLMENT_SETS_QUERY_KEY = "fulfillment_sets" as const
+export const fulfillmentSetsQueryKeys = queryKeysFactory(
+  FULFILLMENT_SETS_QUERY_KEY
+)
+
+export const useDeleteFulfillmentSet = (
+  id: string,
+  options?: Omit<
+    UseMutationOptions<
+      HttpTypes.AdminFulfillmentSetDeleteResponse,
+      FetchError,
+      void
+    >,
+    "mutationFn"
+  >
+) => {
+  return useMutation({
+    mutationFn: () => sdk.admin.fulfillmentSet.delete(id),
+    onSuccess: async (data, variables, context) => {
+      await queryClient.invalidateQueries({
+        queryKey: fulfillmentSetsQueryKeys.detail(id),
+      })
+      await queryClient.invalidateQueries({
+        queryKey: fulfillmentSetsQueryKeys.lists(),
+      })
+
```

Comment:
todo:
```suggestion
  await queryClient.invalidateQueries({
        queryKey: stockLocationsQueryKeys.details(),
      })
```
Add this so when we disable a fulfillment set, the change is reflected in the UI